### DISCUSSION TITLE
java.lang.IllegalArgumentException: No view found for id 0x1 (unknown) for fragment ShellItemRenderer

### DISCUSSION THREAD

**Discussion by jzhao8402** [dotnet/maui#7878] [General]
I'm working on an android app using the Maui AppShell and encounter this very frustrating error. If I freshly start the app, everything works well. But if I use the hardware back button to 'quit' the app and tap the app icon to re-launch it, the app start to crash on global routes when calling Shell.Current.GoToAsync(... The error message is always something like the following.

[AndroidRuntime] java.lang.IllegalArgumentException: No view found for id 0x1 (unknown) for fragment ShellItemRenderer{99fe607} (a37d9ccb-3171-4192-aed4-178335390d1c id=0x1)
[AndroidRuntime] 	at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:513)
[AndroidRuntime] 	at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
[AndroidRuntime] 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)
[AndroidRuntime] 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1758)
[AndroidRuntime] 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)
[AndroidRuntime] 	at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:488)
[AndroidRuntime] 	at android.os.Handler.handleCallback(Handler.java:938)
[AndroidRuntime] 	at android.os.Handler.dispatchMessage(Handler.java:99)
[AndroidRuntime] 	at android.os.Looper.loop(Looper.java:246)
[AndroidRuntime] 	at android.app.ActivityThread.main(ActivityThread.java:8653)
[AndroidRuntime] 	at java.lang.reflect.Method.invoke(Native Method)
[AndroidRuntime] 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
[AndroidRuntime] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)

Not sure if the re-launch restore the route or not. When debug, I saw OnSleep() event fired in App class when I tap the hardware back button, but re-launch the app by tap the app icon, neither OnResume() nor OnStart fired in App class.

Does anyone encounter similar issue? Any help would be appreciated.

**Comment by MartinRothschink** [comment]
I see the the exact same error on Android 11 if I jump between shell pages via shell menu. Crashes happen randomly on a real device. 
I was not yet able to reproduce the issue on an Android emulator.

**Comment by MartinRothschink** [comment]
That's my stacktrace:

java.lang.IllegalArgumentException: No view found for id 0x1 (unknown) for fragment ShellItemRenderer{46ad6a6} (bdd04ca2-ad0c-44cf-a2ef-f2aaa6973711 id=0x1)
androidx.fragment.app.FragmentStateManager.createView FragmentStateManager.java:513
androidx.fragment.app.FragmentStateManager.moveToExpectedState FragmentStateManager.java:261
androidx.fragment.app.FragmentManager.executeOpsTogether FragmentManager.java:1840
androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute FragmentManager.java:1758
androidx.fragment.app.FragmentManager.execPendingActions FragmentManager.java:1701
androidx.fragment.app.FragmentManager$4.run FragmentManager.java:488
android.os.Handler.handleCallback Handler.java:938
android.os.Handler.dispatchMessage Handler.java:99
android.os.Looper.loop Looper.java:246
android.app.ActivityThread.main ActivityThread.java:8633
java.lang.reflect.Method.invoke Method.java
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run RuntimeInit.java:602
com.android.internal.os.ZygoteInit.main ZygoteInit.java:1130

**Comment by AbstractionsAs** [comment]
I have the same problem

**Reply by jzhao8402** [reply]
I have created a repository on github of a simple MAUI project in case anyone wants to check the problem.  

https://github.com/jzhao8402/MauiApp1

The project uses the AppShell with two flyout items for a Login Page and a About Page. The steps to reproduce the problem are quit simple. I have tried on both android emulator or physical devices. 

1.  Launch the app, it works well and you can use the flyout menu to switch the pages. 
2.  Use  the android back button to "close" the app. 
3.  Tap the app icon to relaunch the app, or use the android history button to find you app in the stack and bring it back; the page will be one before you closed the app. 
4.  Use the flyout menu to switch the page, the app will crash with the error:
**Java.Lang.IllegalArgumentException:** 'No view found for id 0x1 (unknown) for fragment ShellItemRenderer
 
Did I miss any critical thing to make it working properly?  I'm frustrating because the project is simple enough yet I cannot fix the issue.  Is this possible a MAUI GA bug?

**Reply by jzhao8402** [reply]
I started to believe this is a MAUI AppShell bug. I have submitted a bug report on the Issues section.

https://github.com/dotnet/maui/issues/7940

**Reply by echannelsolutions** [reply]
I am receiving the same error , any updates on thos`java.lang.IllegalArgumentException: No view found for id 0x13 (unknown) for fragment ShellItemRenderer{73d0a34} (28a214d8-1a53-4780-8c2a-7613b821077f id=0x13)`

How can we fix this ?

**Reply by s-beji** [reply]
I also need to know how we can fix it. 
It's a critical bug for most people who want to build production-ready applications to deploy to a customer.

**Reply by echannelsolutions** [reply]
Okay am able to resolve this - by removing `this.currentitem` or `shell.currentitem` (in appshell.xaml.cs) from my code while constructing the flyout via C#. Hope it helps !

**Reply by douglassimaodev** [reply]
For me this happens when you already have an app with the same identify, so try to uninstall the app from the phone and run vs again

**Comment by wdenejko** [comment]
I have the same issue present in my app on .NET 7 with multiple Android versions. It crashes often, randomly on various devices. Has anyone found any solution/workaround for this issue?

**Reply by Jay-57blocks** [reply]
![image](https://github.com/user-attachments/assets/ca7a2f94-49f3-458f-a111-58701d125767)
I also had this problem, did you solve it please?

**Comment by failwyn** [comment]
I have the same issue with .NET 8...

Microsoft Visual Studio Professional 2022 (64-bit)
Version 17.10.5

**Reply by Jay-57blocks** [reply]
![image](https://github.com/user-attachments/assets/ca7a2f94-49f3-458f-a111-58701d125767)
I also had this problem, did you solve it please?

**Comment by Jay-57blocks** [comment]
![image](https://github.com/user-attachments/assets/ca7a2f94-49f3-458f-a111-58701d125767)
I also had this problem, did you solve it please?

**Reply by s-beji** [reply]
It's a regression bug

**Reply by steffenborup** [reply]
Have you managed to somehow work around it?

**Reply by s-beji** [reply]
Quit use this framework is the best course of action for a serious project for production. Betting on Maui is like betting on silver light. I must say it's a doomed technology and not nearly ready for production it's still in early development and Microsoft is just delusional to think it's useable

**Comment by Jay-57blocks** [comment]
So this is Maui's fault? When will it be fixed? Sometimes the method can
ignore the error for a while?


Best regards/Jay Ren


Sliem Beji ***@***.***>于2024年9月15日 周日下午8:33写道：
